src/main/java/org/mule/tools/apikit/MainAppScaffolder.java

```diff
@@ -44,17 +45,21 @@ public ScaffoldingResult run(ScaffoldingConfiguration config) {
       RAMLFilesParser ramlFilesParser = new RAMLFilesParser(apiFactory, config.getApi());
 
       List<ApikitMainFlowContainer> includedApis = ramlFilesParser.getApisAsList();
+      ResourcesGenerator.replaceReferencesToProperties(config, includedApis);
```

Comment:
why not replace references to properties after ResourceGenerator.generate ? in this case you can replace the references and fail generating the resources

Reply:
It's placed here because the MuleConfigGenerator will then create the httpListenerConfig, otherwise it never "takes" the reference.